src/appindexing.js

```diff
@@ -0,0 +1,130 @@
+'use strict';
+goog.provide('appindexing');
+
+goog.require('goog.json');
+goog.require('utils');
+
+
+appindexing.state = {};
+appindexing.state['androidAppIndexingTagsPresent'] = false;
+appindexing.state['iOSAppIndexingTagsPresent'] = false;
+appindexing.state['androidDetailsComplete'] = false;
+appindexing.state['iOSDetailsComplete'] = false;
+appindexing.state['addediOSTagDom'] = false;
+appindexing.state['addedAndroidTagDom'] = false;
+
+/** Scans through all 'link' tags then updates their 'href' values for those that contain 
+ * "ios-app" or "android-app" to include Branch tracking parameters.
+ * (Stage 1) 
+ */
+appindexing.updateAppIndexingTagsIfPresent = function() {
+	var linkTags = document.getElementsByTagName('link');
+	var numLinkTags = linkTags.length;
+
+	if (!numLinkTags) { return; }
+
+	for (var counter = 0; counter < numLinkTags; counter++) {
+
+		var currLinkTag = linkTags[counter];
+		var currLinkTagHref = currLinkTag.href;
+
+		if (!currLinkTagHref) { continue; }
+
+		if (currLinkTagHref.includes('ios-app')) {
+			appindexing.state['iOSAppIndexingTagsPresent'] = true;
+			currLinkTag.setAttribute('href',  appindexing.addBranchTrackingParams(currLinkTagHref));
+		}
+		if (currLinkTagHref.includes('android-app')) {
+			appindexing.state['androidAppIndexingTagsPresent'] = true;
+			currLinkTag.setAttribute('href',  appindexing.addBranchTrackingParams(currLinkTagHref));
+		}
+	}
+};
+
+/** Checks whether config contains appropriate per platform params. For Android, config needs to contain
+  * both androidPackageName and androidURL. For iOS both, iosAppId and iosURL. Once config is correctly set for
+  * iOS or Android, an App Indexing tag is built and inserted into the webpage.
+  * (Stage 2)
+  */
+appindexing.insertAppIndexingTagsFromConfig = function(options) {
+	if (options.hasOwnProperty('androidPackageName') &&
+		options.hasOwnProperty('androidURL') &&
+		typeof options['androidPackageName'] !== undefined &&
+		typeof options['androidURL'] !== undefined) {
+
+		appindexing.state['androidDetailsComplete'] = true;
+		appindexing.addAppIndexingTag('android', options);
+	}
+	if (options.hasOwnProperty('iosAppId') && 
+		options.hasOwnProperty('iosURL') && 
+		typeof options['iosAppId'] !== undefined && 
+		typeof options['iosURL'] !== undefined) {
+
+		appindexing.state['iOSDetailsComplete'] = true;
```

Comment:
There is different casing in a few different places for ios. We should be consistent. I would just stick with all lowercase